examples/water_boiler/water_boiler.py

```diff
@@ -58,4 +60,7 @@ def update(self, boiler_power, dt):
     plt.xlabel('time')
     plt.ylabel('temperature')
     plt.legend()
-    plt.show()
+    if os.getenv('NO_DISPLAY'):
```

Comment:
Oh, and maybe you could add a comment explaining what this is used for. Since this is an example meant for users to understand, it's important to try to not create unnecessary confusion.